File: tests/test_bridge.js
const { bridgeLore, loreToDict, loreToSql, loreToRust, loreToTypescript } = require('../quilt_bridge/core');

function assert(cond, msg) {
  if (!cond) throw new Error("FAIL: " + msg);
}

// Test loreToDict
const lore = "Line 1\nLine 2";
const d = loreToDict(lore);
assert(d.lore_lines === 2, `lore_lines (got ${d.lore_lines})`);
assert(d.lore_chars === lore.length, `lore_chars (expected ${lore.length} got ${d.lore_chars})`);
assert(d.word_count === 4, `word_count (got ${d.word_count})`);

// Test SQL escape
const sql = loreToSql("It's a 'test'", 1);
assert(sql.includes("''test''"), "sql escape");

// Test Rust escape
const rust = loreToRust('A "test" lore.');
assert(rust.includes('\\"test\\"'), "rust escape");

// Test TypeScript escape
const ts = loreToTypescript('A `${template}` lore.');
assert(ts.includes('\\${template}'), "ts escape");

// Test bridge all
const all = bridgeLore("Test");
assert(typeof all.dict === 'object', "dict");
assert(typeof all.sql === 'string', "sql");
assert(typeof all.rust === 'string', "rust");
assert(typeof all.typescript === 'string', "typescript");

console.log("✓ All 5 JS bridge tests pass");
console.log("  lore_chars:", d.lore_chars, "(expected", lore.length, ")");
console.log("  lore_lines:", d.lore_lines, "(expected 2)");
console.log("  word_count:", d.word_count, "(expected 4)");

File: tests/test_bridge.py
import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..'))
import unittest

from quilt_bridge.core import (
    lore_to_dict, lore_to_sql, lore_to_rust, lore_to_typescript, bridge_lore,
)


class TestBridge(unittest.TestCase):

    def test_dict(self):
        d = lore_to_dict("A scar does not bar entry.\nThe substrate walks.")
        self.assertEqual(d["lore_lines"], 2)
        self.assertEqual(d["lore_chars"], 47)

    def test_sql_escape(self):
        out = lore_to_sql("It's a 'test' lore.", lore_id=1)
        self.assertIn("''test''", out)  # quotes escaped

    def test_rust_escape(self):
        out = lore_to_rust('A "test" lore.')
        self.assertIn('\\"test\\"', out)

    def test_typescript_escape(self):
        out = lore_to_typescript('A `${template}` lore.')
        self.assertIn('\\${template}', out)

    def test_bridge_all(self):
        bridges = bridge_lore("Test lore.")
        self.assertIn("dict", bridges)
        self.assertIn("sql", bridges)
        self.assertIn("rust", bridges)
        self.assertIn("typescript", bridges)


if __name__ == "__main__":
    unittest.main()
